pkg/scheduling/volumeusage.go

```diff
@@ -130,22 +127,32 @@ func resolveDriver(ctx context.Context, kubeClient client.Client, volumeName str
 			return driverName, nil
 		}
 	}
-	if storageClassName != "" {
-		driverName, err := driverFromSC(ctx, kubeClient, storageClassName)
-		if err != nil {
-			return "", err
-		}
-		if driverName != "" {
-			return driverName, nil
+
+	// This state shouldn't be possible during normal operations but if a PV is deleted and its finalizer is removed it is possible.
+	if storageClassName == "" {
```

Comment:
Is this state possible if we fail to resolve the driver to a non-empty string on L122 and then fall down into this conditional statement?

Reply:
Good point, we could potentially remove the empty string check on line 126. I'll need to find out if there's a case where we can't resolve the driver name from a bound volume but we can from it's storage class.

Reply:
I'm actually thinking it may be best to leave this as is, I'm not familiar with the in-tree plugins / storage classes so I don't want to miss an edge case with those so I've just changed the comment. There are a couple of more things I need to validate before I think this is completely ready to merge, if I determine its totally safe to remove I can make the update.